docs/lib/datastore/fragments/flutter/relational/delete-snippet.md

```diff
@@ -1,3 +1,11 @@
 ```dart
-TBD
+(await Amplify.DataStore.query(Post.classType, where: Post.ID.eq("123")))
+    .forEach((element) async {
+  try {
+    await Amplify.DataStore.delete(element);
+    print("Deleted a post");
+  } catch (e) {
+    print("Delete failed: " + e);
+  }
+});
```

Comment:
Do you need to include the outer-most `(` and `)`? 

Instead of doing `forEach` after the `await`, you could use `asStream()` and then iterator over it with the `forEach` on stream.

I wonder if there is a way to merge all of the `delete` calls, too, without that inner `await`.

This might be one way to do it.
```dart
await Amplify.DataStore.query(Post.classType, where: Post.ID.eq("123"))
    .then((queryResults) => {
        Future.forEach(queryResults, (item) => Amplify.DataStore.delete(item))
    })
```

Apologies if I'm bad at Dart. 😳  Syntax might be off.

Reply:
There is no outer `()` The opening parenthesis is ending on the first line itself so we can call `forEach` on the results and not promises and the last `)` is closing the `forEach`

I'm not seeing a benefit of converting these to streams to do the same, this seems more straightforward usage of the APIs. The `then` is not really recommended over `async/await`